### ISSUE TITLE
fix docs with goi 1.66+

### ISSUE THREAD

**Issue by gcsideal** [libvips/libvips#1836]
**VIPS v8.10.1 fails to build on Debian/Sid**
"Namespace conflict: DocSection('rect')" error during build.

**To Reproduce**
1. On Debian/Sid: 'apt source vips'
2. Do 'dpkg-buildpackage' in vips-8.10.1 directory.

**Actual behavior**
During the generation of Vips-8.0.gir:
Traceback (most recent call last):
  File "/usr/bin/g-ir-scanner", line 99, in <module>
    sys.exit(scanner_main(sys.argv))
  File "/usr/lib/x86_64-linux-gnu/gobject-introspection/giscanner/scannermain.py", line 606, in scanner_main
    main.transform()
[...]
  File "/usr/lib/x86_64-linux-gnu/gobject-introspection/giscanner/ast.py", line 461, in append
    raise ValueError("Namespace conflict: %r" % (node, ))
ValueError: Namespace conflict: DocSection('rect')
make[3]: *** [/usr/share/gobject-introspection-1.0/Makefile.introspection:156: Vips-8.0.gir] Error 1

Full build log: http://qa-logs.debian.net/2020/09/26/vips_8.10.1-1_unstable.log

**Environment**
 - OS: up to date Debian/Sid on amd64
 - Vips: 8.10.1
 - Python: 3.18.6
 - gobject-introspection: 1.66.0

**Comment by jcupitt** [comment]
Hello László,

It's working on ubuntu 20.04 for me with g-o-i 1.64.1-1, so I guess there's some recent change there. I'll try with 1.66.

**Comment by jcupitt** [comment]
I see the error here with 1.67. I added these prints to `gobject-introspection/giscanner/ast.py:`:

```python
    def append(self, node, replace=False):
        previous = self.names.get(node.name)
        if previous is not None:
            if not replace:
                print(f"duplicate {node}")
                print(f"node.doc_position = {node.doc_position}")
                print(f"previous.doc_position = {previous.doc_position}")
                raise ValueError("Namespace conflict: %r" % (node, ))
            self.remove(previous)

        self.track(node)
        self.names[node.name] = node
```

Which reveal that g-o-i functions and sections now share the same namespace. This means you can't have a function called `vips_error()`, and a documentation section called `error`.

This seems a bit crazy to me (???) but I suppose we have to adapt. I've renamed the doc sections error, rect and interpolate and the docs now build for me in 8.10 HEAD.

**Comment by jcupitt** [comment]
I opened an issue on g-o-i: https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/359

**Comment by jcupitt** [comment]
It's building now, but some doc sections are missing due to the conflict.

Let's fix this properly during 8.11. I'll leave the bug open.

**Comment by gcsideal** [comment]
> Let's fix this properly during 8.11. I'll leave the bug open.

Do you have any plans when that release is going to happen?

**Comment by jcupitt** [comment]
Some time early next year for 8.11 I guess. 

The head of the 8.10 branch should be working now, is there still an issue there? I'll do an 8.10.2 with this fix within a week.

**Comment by gcsideal** [comment]
> Some time early next year for 8.11 I guess.

I see. That would be late for the next Debian stable release I believe.

> The head of the 8.10 branch should be working now, is there still an issue there? I'll do an 8.10.2 with this fix within a week.

Need to check the ARM64 and PPC64EL self-testing regression soon then.

It is working in the sense of building, but quoting you:
> It's building now, but some doc sections are missing due to the conflict.

**Comment by jcupitt** [comment]
Should be fixed in 8.10.2 https://github.com/libvips/libvips/releases/tag/v8.10.2
